### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadInterruptedException
  id: ThreadInterruptedException
  children:
  - System.Threading.ThreadInterruptedException.#ctor
  - System.Threading.ThreadInterruptedException.#ctor(System.String)
  - System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException
  fullName: System.Threading.ThreadInterruptedException
  type: Class
  summary: "時擲回的例外狀況<xref href=&quot;System.Threading.Thread&quot;></xref>處於等候狀態時中斷。"
  remarks: "在建立的執行緒之後，就在一或多個<xref:System.Threading.ThreadState>指出它終結之前有效。</xref:System.Threading.ThreadState> 呼叫<xref:System.Threading.Thread.Interrupt%2A>當執行緒處於<xref:System.Threading.ThreadState>的狀態會造成在目標執行緒中擲回 ThreadInterruptedException。</xref:System.Threading.ThreadState> </xref:System.Threading.Thread.Interrupt%2A> 如果執行緒不在<xref:System.Threading.ThreadState>狀態，直到執行緒會進入該狀態不擲回例外狀況。</xref:System.Threading.ThreadState> 如果執行緒永不封鎖，它無法完成沒有插斷。       ThreadInterruptedException 會使用 HRESULT COR_E_THREADINTERRUPTED，0x80131519 的值。       如需 ThreadInterruptedException 的執行個體的初始屬性值的清單，請參閱<xref:System.Threading.ThreadInterruptedException.%23ctor%2A>建構函式。</xref:System.Threading.ThreadInterruptedException.%23ctor%2A>"
  example:
  - "The following code example shows the behavior of a running thread when it is interrupted and subsequently gets blocked.  \n  \n [!code-cpp[System.Threading.Thread.Interrupt#1](~/add/codesnippet/cpp/t-system.threading.threa_1_1.cpp)]\n [!code-vb[System.Threading.Thread.Interrupt#1](~/add/codesnippet/visualbasic/t-system.threading.threa_1_1.vb)]\n [!code-cs[System.Threading.Thread.Interrupt#1](~/add/codesnippet/csharp/t-system.threading.threa_1_1.cs)]"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadInterruptedException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新執行個體<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>使用預設屬性的類別。"
  remarks: "下表顯示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>的執行個體的初始屬性值      |屬性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|當地語系化的錯誤訊息字串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException ();
    parameters: []
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新執行個體<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>類別指定的錯誤訊息。"
  remarks: "下表顯示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>的執行個體的初始屬性值      |屬性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|錯誤訊息字串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message);
    parameters:
    - id: message
      type: System.String
      description: "解釋例外狀況的原因的錯誤訊息。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新執行個體<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>使用序列化資料的類別。"
  remarks: "重新建構所需的資料流上傳輸的例外狀況物件的還原序列化期間呼叫這個建構函式。 如需詳細資訊，請參閱[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected ThreadInterruptedException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>其中保存有關擲回例外狀況的序列化的物件資料。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>其中包含關於來源或目的端內容資訊。</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新執行個體<xref href=&quot;System.Threading.ThreadInterruptedException&quot;></xref>類別指定的錯誤訊息以及造成此例外狀況的內部例外狀況的參考。"
  remarks: "做為前一個例外狀況直接結果應該包括在前一個例外狀況的參考就會擲回的例外狀況<xref:System.Exception.InnerException%2A>屬性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>屬性會傳回相同的值傳遞到建構函式，或`null`如果<xref:System.Exception.InnerException%2A>屬性沒有提供內部例外值給建構函式。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表顯示<xref:System.Threading.ThreadInterruptedException>.</xref:System.Threading.ThreadInterruptedException>的執行個體的初始屬性值      |屬性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|內部例外狀況參考。 |  |<xref:System.Exception.Message%2A>|錯誤訊息字串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadInterruptedException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "解釋例外狀況的原因的錯誤訊息。"
    - id: innerException
      type: System.Exception
      description: "造成目前例外狀況的例外狀況。 如果`innerException`參數不是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>中, 引發目前的例外狀況<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>處理內部例外狀況區塊。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadInterruptedException.#ctor
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadInterruptedException.#ctor*
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException.ThreadInterruptedException
